go/flowctl/cmd-discover.go

```diff
@@ -286,6 +309,20 @@ func writeConfigStub(ctx context.Context, image string, w io.WriteCloser) error
 	return nil
 }
 
+// getDefaultType returns the type to use for generating a default value for endpoint configuration.
+// It will always prefer a scalar type if a location allows multiple types.
+func getDefaultType(inference *pf.Inference) string {
```

Comment:
Is this effectively picking `"string"` from `["string", "null"]`? Or do we have cases where options are `["object", "string"]` messes?

Reply:
I think in practice pulling `"string"` from `["string", "null"]`, but `source-gcs` does have a `googleCredentials` key that takes an object (gcp service account keys are JSON objects).